### PR TITLE
3 remove gyms

### PR DESCRIPTION
**Pull request by Scott-Borecki**
### Changes Implemented:
  - Update database schema for more simplified and streamlined flow (see image below)
    - Remove Gym and Invitation tables
    - Yelp API now "acts" as Gym table
    - Rename GymMembers table to GymMemberships
    - Events are now created through GymMemberships
    - `gym_name` column added to GymMemberships table
  - Updated tests and code to work with the updated schema
  - Handrolled GymSerializer

### Quality Control Checklist:
  - [X] Code adheres to Rubocop styling (if unavoidable infractions, please clarify below)
  - [X] 100% SimpleCov test coverage (if below, please clarify below)
  - [X] Last commit passed TravisCI Branch Build check

### Blockers (if applicable):
  - None

### Next Steps & Additional Notes:
  - We should add an endpoint for Gym Show, using the Yelp API
  - Need to test hand-rolled GymSerializer, along with other serializers too!
  - Review and update project cards
  - Update JSON endpoint documentation
  - Review and update FE endpoint calls
 
## Updated Database Schema
image

### CODE REVIEW

**Review by tvaroglu — APPROVED** [review]
Forklift of the century right there.. thank you so much for smashing through all that.... I really do think this makes for cleaner architecture and will drastically simplify things!  I've got some big shoes to fill on this next rotation!!

File: db/migrate/20210919213448_drop_gyms.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class DropGyms < ActiveRecord::Migration[5.2]
+  def change
+    remove_foreign_key :gym_members, :gyms
+    remove_foreign_key :events, :gyms
+    drop_table :gyms
+  end
+end
```
Comment (carolinectan) [inline db/migrate/20210919213448_drop_gyms.rb:7]: Nice work using migrations vs. dropping db tables

**Comment by ealwafai** [comment]
Great work!

**Review by gunnarrunner — APPROVED** [review]
Awesome. First time looking through the backend and yeahh this looks great. Very easy to read and to follow. Thank you for making it look pretty.

**Comment by bfl3tch** [comment]
Updated schema is awesome

**Review by carolinectan — APPROVED** [review]
Amazing work adding the new migrations to update our schema and updating all of the code to reflect those changes! THANK YOU!!

File: config/routes.rb [hunk 2]
```diff
@@ -4,13 +4,11 @@
...
       resources :gym_search, only: [:index]
-      delete 'users/:user_id/gyms/:gym_id', to: 'users/gyms/gym_members#destroy'
       resources :users, only: [:show, :update, :create] do
         resources :events, only: [:index], controller: 'users/events'
         resources :friendships, only: [:index, :create, :destroy], controller: 'users/friendships'
-        resources :gyms, only: [:index], controller: 'users/gyms' do
-          resources :events, only: [:create], controller: 'users/gyms/events'
-          resources :gym_members, only: [:create], controller: 'users/gyms/gym_members'
+        resources :gym_memberships, only: [:index, :create, :destroy], controller: 'users/gym_memberships' do
+          resources :events, only: [:create], controller: 'users/gym_memberships/events'
```
[the remark this answers is not in the document]
Comment (bfl3tch) [reply config/routes.rb:11]: For sure, much cleaner

File: db/migrate/20210919213448_drop_gyms.rb (hunk 1 above)
Comment (bfl3tch) [reply db/migrate/20210919213448_drop_gyms.rb:7]: 100% - hadn't seen this before as an option

**Review by bfl3tch — APPROVED** [review]
Excellent work

**Review by ealwafai — APPROVED** [review]
Good to go!

**Pull request merged**